docs/en/explanation/implicit-indexes.rst

```diff
@@ -0,0 +1,62 @@
+Implicit indexes
+================
+
+Ever noticed the DBAL creating indexes you did not remember asking for,
+with names such as ``IDX_885DBAFAA76ED395``? In this document, we will
+distinguish three types of indexes:
+
+user-defined indexes
+  indexes you did ask for
+
+DBAL-defined-indexes
+  indexes you did not ask for, created on your behalf by the DBAL
+
+RDBMS-defined indexes
+  indexes you did not ask for, created on your behalf by the RDBMS
+
+Implicit indexes can be created by some database platforms when you
```

Comment:
Purely from the terminology standpoint, it's not obvious that the latter two are the implicit ones.

Reply:
Plus in this sentence I'm really talking about the RDBMS-defined ones :sweat_smile: